blog/2020-07-01-LetsOptaPlanYourJbpmTasksPart1.adoc

```diff
@@ -0,0 +1,215 @@
+= Let's OptaPlan your jBPM tasks (part 1) - Integrating the two worlds
+:page-interpolate: true
+:awestruct-author: wmedvede
+:awestruct-layout: blogPostBase
+:awestruct-tags: [feature, task assigning, bpm integration]
+
+:awestruct-share_image_filename: CreditDisputeProcessDiagram.png
+
+Traditionally, business process management systems (BPMS) assign human tasks to users on a group basis.
+This information is part of the human tasks configuration.
+
+In the following process, we can see how the Resolve Dispute and Notify Customer tasks are configured with the
+CreditAnalyst and ClientRelations groups respectively:
+
+image::CreditDisputeProcessDiagram.png[]
+
+At runtime, every time a new instance of one of these tasks is created it is automatically assigned to the respective
+group with the status Ready.
+
+This makes the task automatically available to all users that belong to the group.
+All users in the group  see it in their respective *Task Inbox* but none are the actual owner or responsible for the task.
+To be able to work with a particular task, a user must execute the claim operation to reserve the task for that user.
+This is a completely manual procedure.
+
+The following image shows a list of tasks with the Ready status:
+
+image::TaskList-NoTaskIsSelected.png[]
+
+The following image shows the result of the claim operation executed on task #74 by the user katy:
+
+image::TaskList-UserKatyClaimedATask.png[]
+
+Although this is the standard procedure for most BPMS products, it is not flexible enough for all use cases, for example:
+
+* Scenarios with a huge set of tasks that require an optimal distribution of tasks between users by a specific criteria.
+For example, distribute the tasks fairly between users, minimize the time to complete all the tasks, consider the tasks priorities, and so forth.
+
+* Scenarios where it's desired to combine business data for making the tasks assignments, because BPMS engines normally
+assign tasks on a group basis but don't have the ability to include business data for making the assigning decisions.
+
+* When instant reaction is required, new tasks are created or the existing tasks are completed. All of these events might
+require a redistribution of the work to ensure the assignments are still optimal.
+
+So in general, when using standard BPMS products, users might find a large list of tasks up-front but do not have the
+knowledge to manually claim and execute them in the best or optimal way. And the responsibility for picking the tasks in
+the proper order is delegated to the users.
+
+== How about asking OptaPlanner to do some of this work?
+
+In the last months we've been working on integrating https://www.optaplanner.org/[OptaPlanner] and https://jbpm.org/[jBPM]  with that purpose.
+
+The concept is very simple: *"_let each tool do the piece of work it is good at, integrate the results, and provide a
```

Comment:
(soft suggestion)
Avoid saying that something is simple or easy, especially if it's a complex topic.
If people don't immediately get it, that makes one feel as not part of the target audience, so this subject isn't meant for them.

Reply:
maybe that doesn't avoidance doesn't apply here, reading this now in detail.

Reply:
yes, when I'm saying "simple" I'm referring to the "simple idea of integrating the two worlds"